packages/swingset/src/get-nav-tree.ts

```diff
@@ -83,5 +84,29 @@ export function getNavigationTree(
   }
 
   const tree = Array.from(categories.values())
+
+  tree.sort(compareTitleSort)
+
   return tree
 }
+
+function compareTitleSort<T extends CategoryNode | NavigationNode>(
+  a: T,
+  b: T
+): -1 | 0 | 1 {
+  const aHasChildren = 'children' in a && a.children.length > 0
+  const bHasChildren = 'children' in b && b.children.length > 0
+  if (aHasChildren) {
+    a.children.sort(compareTitleSort)
+  }
+  if (bHasChildren) {
+    b.children.sort(compareTitleSort)
+  }
```

Comment:
I'm thinking this might not be the proper place to call sort recursively, as each element might get passed to the compare function more than once, which would result in us sorting an element's children multiple times. We'll likely want to separate out traversing the tree from the actual sorting.

Reply:
Are you suggesting, something like:

Create a toBeSorted = Map()
Traverse the rest tree, while checking which nodes have children
If hasChildren == true. Store reference to the node in toBeSorted
Loop through the map and sort the children

Reply:
Are you suggesting, something like:

- Create a `toBeSorted = Map()`
- Traverse the tree, while checking which nodes have children
- If `hasChildren == true`. Store reference to the node in `toBeSorted`
- Loop through the map and sort